### ISSUE TITLE
Builtin functions should work with a `.` in the name

### ISSUE THREAD

**Issue by hugomg** [pallene-lang/pallene#221]
For example, instead of `io_write` and `math_sqrt` we should be able to write `io.write` and `math.sqrt`. Doing this for builtins will pave the way for allowing the same for non built-in modules in the future.

**Comment by hugomg** [comment]
Today I went through the code to develop a plan of how we could implement this. I think I have a basic plan that might work.

For the parse we don't need to do anything. Something like `io.write` can continue to be parsed as an `ast.Var.Dot`.

```
{
    _tag = "ast.Var.Dot",
    exp = {
       _tag = "ast.Exp.Var", 
       var = {  _tag = "ast.Var.Name",  name = "io" }  },
    name = "write"
}
```

The type checker could then transform that `ast.Var.Dot` into a flat `ast.Var.Name`. This way the `to_ir` and the `coder` won't need to be modified and the changes can be restricted to the type checker.

```
{
    _tag = "ast.Var.Name",
    name = "io.write"
}
```

To do this, we will need to change the behavior of the `check_var` method. Instead of only annotating the var node with a `_type` and `_name` field, it will also have to return the (potentially modified) var node, similarly to how check_exp does it. That is, instead of just doing a `self:check_var(exp.var)`, the callers will have to do a `exp.var = self:check_var(exp.var)`.

We might want to add a `Module` case to the `checker.Name` tagged union, to represent names such as `io` and `string` that correspond to a module name. These names should only be allowed to be part of a Var.Dot. If someone tries to do `local x = io` then it should be a compile-time error, similarly to what happens when trying to assign a type name to a local variable.

I think this should cover most of it but if something unexpected happens then we can figure it out.

**Comment by hugomg** [comment]
Implemented in #224
